internal/fields/validate.go

```diff
@@ -161,51 +170,56 @@ func createValidatorForDirectoryAndPackageRoot(fieldsParentDir string, finder pa
 	v.allowedCIDRs = initializeAllowedCIDRsList()
 
 	fieldsDir := filepath.Join(fieldsParentDir, "fields")
-	v.Schema, err = loadFieldsFromDir(fieldsDir)
-	if err != nil {
-		return nil, fmt.Errorf("can't load fields from directory (path: %s): %w", fieldsDir, err)
-	}
 
-	if v.disabledDependencyManagement {
-		return v, nil
+	var fdm *DependencyManager
+	if !v.disabledDependencyManagement {
+		packageRoot, found, err := finder.FindPackageRoot()
+		if err != nil {
+			return nil, fmt.Errorf("can't find package root: %w", err)
+		}
+		if !found {
+			return nil, errors.New("package root not found and dependency management is enabled")
+		}
+		fdm, v.Schema, err = initDependencyManagement(packageRoot, v.specVersion, v.enabledImportAllECSSchema)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize dependency management: %w", err)
+		}
 	}
 
-	packageRoot, found, err := finder.FindPackageRoot()
+	fields, err := loadFieldsFromDir(fieldsDir, fdm, v.injectFieldsOptions)
 	if err != nil {
-		return nil, fmt.Errorf("can't find package root: %w", err)
-	}
-	// As every command starts with approximating where is the package root, it isn't required to return an error in case the root is missing.
-	// This is also useful for testing purposes, where we don't have a real package, but just "fields" directory. The package root is always absent.
-	if !found {
-		logger.Debug("Package root not found, dependency management will be disabled.")
-		v.disabledDependencyManagement = true
-		return v, nil
+		return nil, fmt.Errorf("can't load fields from directory (path: %s): %w", fieldsDir, err)
 	}
 
-	bm, ok, err := buildmanifest.ReadBuildManifest(packageRoot)
+	v.Schema = append(fields, v.Schema...)
```

Comment:
At this point v.Schema would just contain the fields retrieved from the ECS reference?

Reply:
Correct, it would contain the ECS schema fields when import all ECS schema fields is enabled.